src/main/java/de/tum/in/www1/artemis/web/rest/ResultResource.java

```diff
@@ -276,4 +278,59 @@ public ResponseEntity<Result> createResultForExternalSubmission(@PathVariable Lo
         return ResponseEntity.created(new URI("/api/results/" + savedResult.getId()))
                 .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, ENTITY_NAME, savedResult.getId().toString())).body(savedResult);
     }
+
+    /**
+     * GET /exercises/:exerciseId/feedback-details : get all feedback details and participations for an exercise.
+     *
+     * @param exerciseId The ID of the exercise
+     * @return A response entity containing feedback details and participations
+     */
+    @GetMapping("exercises/{exerciseId}/feedback-details")
+    @EnforceAtLeastTutor
+    public ResponseEntity<FeedbackDetailsResponse> getAllFeedbackDetailsForExercise(@PathVariable Long exerciseId) {
+        log.debug("REST request to get all Feedback details for Exercise {}", exerciseId);
+        Exercise exercise = exerciseRepository.findByIdElseThrow(exerciseId);
+
+        // Get all Student participations for the exercise
+        Set<StudentParticipation> participation;
+        if (exercise.isTeamMode()) {
+            participation = studentParticipationRepository.findByExerciseIdWithLatestAndManualResultsWithTeamInformation(exercise.getId());
+        }
+        else {
+            participation = studentParticipationRepository.findByExerciseIdWithLatestAndManualResultsAndAssessmentNote(exercise.getId());
+        }
+        removeSubmissionAndExerciseData(participation);
+
+        List<Feedback> allFeedback = new ArrayList<>();
+
+        // For each participation, get the feedback from its latest result
+        for (StudentParticipation singleParticipation : participation) {
+            Result latestResult = singleParticipation.getResults().stream().max(Comparator.comparing(Result::getCompletionDate)).orElse(null);
+
+            if (latestResult != null) {
+                List<Feedback> feedback = getResultDetails(singleParticipation.getId(), latestResult.getId()).getBody();
+                if (feedback != null) {
+                    allFeedback.addAll(feedback);
+                }
+            }
+        }
```

Comment:
Why don't we retrieve all the latest results in one SQL query? 

We are now saving the client side quite some time but this still seems to be quite a number of separate SQL queries to the database to me. I think we should not have more than 1-2 database queries for this endpoint (if possible).

Reply:
i fixed it further, now it should be fine. Thanks for the feedback 👍 😁